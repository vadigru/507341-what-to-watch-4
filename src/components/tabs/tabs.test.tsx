import * as React from "react";
import * as renderer from "react-test-renderer";
import Tabs from "./tabs";

const tabNames = [`Overview`, `Details`, `Reviews`];

it(`Should render Tabs component`, () => {
  const changeGenre = jest.fn();
  const showDefaultMovies = jest.fn();

  const tree = renderer
    .create(
        <Tabs
          className={`catalog__genres-`}
          tabNames={tabNames}
          activeTab={tabNames[0]}
          onTabClick={changeGenre}
          onGenreTabClick={showDefaultMovies}
        />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
