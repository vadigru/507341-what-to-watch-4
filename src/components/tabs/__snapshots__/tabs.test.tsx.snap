// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tabs component 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Overview
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Details
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Reviews
    </a>
  </li>
</ul>
`;
